### PR TITLE
Updated the test homework method and docs

### PR DESCRIPTION
**Pull request by AnnaDinaburgVulikh**
Hi @sleviim ,
It's the first part of the long pr.
This one includes only the ammended test logic for students homework.

### CODE REVIEW

File: docs/course_materials/exercises/test_exercises/conftest.py [hunk 1]
```diff
@@ -24,38 +26,47 @@ def home_folder():
...
-                self.test_answers(self.expected_stdout, student_stdout.strip(),
-                                  word_pattern=self.exact_answer)
+                student_work += self.run_cmd()
+                LOGGER.debug(f'run CMD, {student_work}')
+        
+        # Reads code file
+        if self.expected_pycode:
+            student_work += self.get_student_code()
+
+        # testing the code
```
Comment (sleviim) [inline docs/course_materials/exercises/test_exercises/conftest.py:61]: testing -> Testing

File: docs/course_materials/exercises/test_exercises/conftest.py [hunk 2]
```diff
@@ -98,25 +111,49 @@ def test_answers(expected_list, answer_list, word_pattern=False):
...
-                if re.search(pattern, line, re.MULTILINE):
-                    LOGGER.debug("MATCHED")
-                    matched = True
-                    break
+            LOGGER.debug(f'answers: {answers}')
+            matched = Test_helpers.test_answer_match(pattern, answers)
             if not matched:
-                message += '\n' + answer[1]
-        assert len(message) == 0, LOGGER.warning(message)
+                if isinstance(answer[1], list):
```
Comment (sleviim) [inline docs/course_materials/exercises/test_exercises/conftest.py:124]: isinstance -> isInstance

File: docs/course_materials/exercises/test_exercises/conftest.py [hunk 3]
```diff
@@ -24,38 +26,47 @@ def home_folder():
...
+        if self.expected_pycode:
+            student_work += self.get_student_code()
+
+        # testing the code
+        LOGGER.info('Testing the code...')
+        LOGGER.debug(student_work)
+        test_code, message = self.test_answers(self.tests_list, student_work)
+        LOGGER.debug(message)
+        assert test_code, LOGGER.info('Good job, ' +
+                                      f'but needs some work:\n{message[:-1]}')
```
Comment (sleviim) [inline docs/course_materials/exercises/test_exercises/conftest.py:67]: work -> improvments or fixes

**Review by sleviim — CHANGES_REQUESTED** [review]
Great work, thanks!
Added some grammar fixes.

I think it would be great if you'll attach a short demo/screenshots for one of the tests you've made.
You can upload it to https://imgur.com/ so it will be available as a link.
What do you think?

**Review by sleviim — COMMENTED** [review]
remove blanks - Approved :)
Let's wait for the other patch so those would merge together.

File: docs/course_materials/exercises/test_exercises/conftest.py (hunk 2 above)
Comment (AnnaDinaburgVulikh) [reply docs/course_materials/exercises/test_exercises/conftest.py:124]: it's a builtin function

**Review by sleviim — APPROVED** [review]

**Pull request merged**